### ISSUE TITLE
Resize events for dynamically added elements using data-interchange not working

### ISSUE THREAD

**Issue by JarvisH** [zurb/foundation-sites#10340]
#### How to reproduce this bug:

1. Add image with data-interchange attr. after main $(document).foundation() call
2. Call ".foundation()" on added element or document
2. Resize window

#### What should happen:
Image src should change

#### What happened instead:
Nothing

#### Cause:
The interchange plugin now listens for resize event directly on element instead of on window.
This was changed in "Replace global resize handler in interchange with zf trigger #9626"

Interchange adds data-resize attr. to elements, but at that point util triggers has already been added.

```
// foundation.interchange.js

    /**
   * Initializes the Interchange plugin and calls functions to get interchange functioning on load.
   * @function
   * @private
   */
  _init() {
    MediaQuery._init();

    var id = this.$element[0].id || GetYoDigits(6, 'interchange');
    this.$element.attr({
      'data-resize': id,
      'id': id
    });

    this._addBreakpoints();
    this._generateRules();
    this._reflow();
  }

```

```
// foundation.util.triggers.js

Triggers.Initializers.addResizeListener = function(debounce){
  let $nodes = $('[data-resize]');
  if($nodes.length){
    debounceGlobalListener(debounce, 'resize.zf.trigger', Triggers.Listeners.Global.resizeListener, $nodes);
  }
}
```

**Comment by hikouki** [comment]
Proposal the following.

```js
new Foundation.Interchange($('[data-interchange]'));
```

**Comment by kball** [comment]
I think this gets to a key problem with how the triggers module is currently implemented - it expects a global parse once to understand targets. I'd love to figure out how to make it easily and automatically update with new targets... I'm thinking that since we now have the explicit dependency, we should be able to create a callback or something similar... thoughts @coreysyms?

**Comment by coreysyms** [comment]
@JarvisH did you create a new instance of interchange? 
var elem = new Foundation.Interchange(element, options);

**Comment by JarvisH** [comment]
@coreysyms no, I did not use Interchange programmatically, just via the main .foundation() call and data-interchange attributes.

**Comment by alsodenn** [comment]
I try to realize a lazy-load with interchange. I have the same problem as described above.
@coreysyms  "new Foundation.Interchange(element, options)" does not work too.

Here a Codepen to see the problem:
https://codepen.io/alsodenn/pen/WZomJP

If you resize the browser the interchange will not replace the sources.

Is there a solution to this problem? 
Will this bug be fixed soon?

**Comment by SassNinja** [comment]
I'm joining the discussion because I'm facing exactly the same issue:
If I initialize an interchange image programmatically it doesn't work on resize.

@JarvisH your mentioned temporary fix by adding `data-resize` doesn't seem to work. The issue remains for me.

However what is working is to manually call the `_reflow` method in a custom changed mediaquery event listener.
I've created an extensive codepen that shows the issue and the quick fix I'm talking about:
https://codepen.io/KaiTheGreat/pen/Qqvqvj

That quick fix is of course not a permanent solution for me so I tried to figure more about this issue but I'm somehow stuck in the triggers util.
@kball would you help me understand how this is working and what's the source of the issue?

I was able to fix it in the interchange events method by replacing resizeme.zf.trigger listener with
```javascript
$(window).on('changed.zf.mediaquery.'+this.id, () => ...
```
But this doesn't feel right...

**Comment by blakestein** [comment]
I had previously had this issue when using the Interchange component in this way:

`
new Foundation.Interchange($('[data-interchange]'));
`

But as of 6.5.0 the source does appear to swap correctly.

**Comment by jesserosenfield** [comment]
I'm having the same issue using swup.js to render pages with AJAX

```
swup.on('contentReplaced', function() {
     // doesn't work
     $(document).foundation();

     // doesn't work
     Foundation.reInit();

     // doesn't work
     Foundation.reInit('mediaquery');

     // doesn't work
     Foundation.reInit('interchange');

     // doesn't work
     Foundation.reInit(['mediaquery', 'interchange']);

     // doesn't work
     Foundation.reInit($('[data-interchange]'));
```
